### ISSUE TITLE
DSM generator with a building shp

### ISSUE THREAD

**Issue by atoulou** [UMEP-dev/UMEP#613]
Hello,

I would like to generate a DSM with the preprocessor of UMEP on a part of the Lyon metropolis. 
I have a DEM file (1m resolution), and a building vector file with a "HAUTEUR" column where the building heights are.
When I generate the DSM, not all the buildings are taken into account (although the building heights are available in the building vector file).

Why is this happening ?

Thanks in advance for your help,

Best regards

**Desktop 

 - OS: [Windows 64 bits]
 - QGIS Version [3.34.4]
 - UMEP Version [4.0.4] and UMEP for processing : 2.0.13

[DSM_generator_pb.zip](https://github.com/UMEP-dev/UMEP/files/15320011/DSM_generator_pb.zip)

![DSM_result](https://github.com/UMEP-dev/UMEP/assets/158269725/d6c9e2d9-b09e-4476-8f8a-08f3cb9e6b6e)
![buildings_with_height](https://github.com/UMEP-dev/UMEP/assets/158269725/f83a6031-a31a-4f51-b7f4-d0659148fafb)
![DSM_window_for_generation](https://github.com/UMEP-dev/UMEP/assets/158269725/73665367-784d-45fe-9ad5-8c40d09f4748)

**Comment by nilswallenberg** [comment]
Hi,

Hmm, not sure. I could not replicate your error. Here's my result:
[DSM.zip](https://github.com/UMEP-dev/UMEP/files/15320573/DSM.zip)

Cheers,
Nils

**Comment by atoulou** [comment]
Actually your file is good, but I ran the procedure again and I getthe same result (without all the buildings), did you do something different ? Or do you have any idea what can I check that might cause this error ?

Thanks again

Best regards

Aurore

**Comment by nilswallenberg** [comment]
No, used same settings as you did. Strange that it adds only some buildings. Not sure what could cause this.

**Comment by biglimp** [comment]
This looks like an upstream issue. I will close re-open if necessary.
